numpy/core/tests/test_overrides.py

```diff
@@ -425,3 +427,151 @@ def __array_function__(self, func, types, args, kwargs):
         # note: the internal implementation of np.sum() calls the .sum() method
         array = np.array(1).view(MyArray)
         assert_equal(np.sum(array), 'summed')
+
+
+class TestArrayLike:
+
+    class MyArray():
+
+        def __init__(self, function=None):
+            self.function = function
+
+        def __array_function__(self, func, types, args, kwargs):
+            try:
+                my_func = getattr(TestArrayLike.MyArray, func.__name__)
+            except AttributeError:
+                return NotImplemented
+            return my_func(*args, **kwargs)
+
+    def add_method(name, enable_value_error=False):
+        def _definition(*args, **kwargs):
+            # Check that `like=` isn't propagated downstream
+            assert 'like' not in kwargs
+
+            if enable_value_error and 'value_error' in kwargs:
+                raise ValueError
+
+            return TestArrayLike.MyArray(getattr(TestArrayLike.MyArray, name))
+        setattr(TestArrayLike.MyArray, name, _definition)
+
+    @pytest.mark.parametrize('function, args, kwargs', [
+        ('array',
+         (1,),
+         {}),
```

Comment:
A helper like the following makes this more readable:
```
def args(*args, **kwargs):
    return args, kwargs
```
used as
```
('require', *func_args(np.arange(6).reshape(2, 3), requirements=['A', 'F'])),
```

Reply:
Done in https://github.com/numpy/numpy/pull/16935/commits/a8ef83a7454ba06c1bbfeaf4ade66cc397fe44fc .